### PR TITLE
feat(bot,types)!: Add partial guild to interaction

### LINKED ISSUE #3600: [api-docs] bring over guild partial (#6870)
A new commit was made into the api-docs repo: https://github.com/discord/discord-api-docs/commit/29c34ce6360523fcbe4ee7fdf73adb2022090efb

This is a bot created issue.

### PR DESCRIPTION
**Pull request by Fleny113**
Add the `guild` field to DiscordInteraction as a partial guild.

This is a breaking change since the interaction transformer (and customizer) now takes a different payload object as argument to provide the `shardId` for the guild transformer that is now required

- Upstream: https://github.com/discord/discord-api-docs/pull/6870
- fixes #3600

### CODE REVIEW

File: packages/bot/src/transformers/interaction.ts [hunk 1]
```diff
@@ -220,55 +223,62 @@ const baseInteraction: Partial<Interaction> & BaseInteraction = {
...
+  if (props.id && payload.interaction.id) interaction.id = bot.transformers.snowflake(payload.interaction.id)
+  if (props.applicationId && payload.interaction.application_id)
+    interaction.applicationId = bot.transformers.snowflake(payload.interaction.application_id)
+  if (props.type && payload.interaction.type) interaction.type = payload.interaction.type
+  if (props.token && payload.interaction.token) interaction.token = payload.interaction.token
+  if (props.version && payload.interaction.version) interaction.version = payload.interaction.version
+  if (props.locale && payload.interaction.locale) interaction.locale = payload.interaction.locale
+  if (props.guildLocale && payload.interaction.guild_locale) interaction.guildLocale = payload.interaction.guild_locale
+  if (props.guild && payload.interaction.guild)
+    // @ts-expect-error payload.interaction.guild is a Partial<DiscordGuild>
```
Comment (lts20050703) [inline packages/bot/src/transformers/interaction.ts:244]: Are we sure that it's okay to pass Partial stuff to the transformer?

File: packages/bot/src/transformers/interaction.ts [hunk 2]
```diff
@@ -220,55 +223,62 @@ const baseInteraction: Partial<Interaction> & BaseInteraction = {
...
+      componentType: payload.interaction.data.component_type,
+      customId: payload.interaction.data.custom_id,
+      components: payload.interaction.data.components?.map((component) => bot.transformers.component(bot, component)),
+      values: payload.interaction.data.values,
+      id: payload.interaction.data.id ? bot.transformers.snowflake(payload.interaction.data.id) : undefined,
+      name: payload.interaction.data.name,
+      resolved: payload.interaction.data.resolved ? transformInteractionDataResolved(bot, payload.interaction.data.resolved, guildId) : undefined,
+      options: payload.interaction.data.options?.map((opt) => bot.transformers.interactionDataOptions(bot, opt)),
+      targetId: payload.interaction.data.target_id ? bot.transformers.snowflake(payload.interaction.data.target_id) : undefined,
+      // guildId: payload.interaction.data.guild_id ? bot.transformers.snowflake(payload.interaction.data.guild_id) : undefined,
```
Comment (lts20050703) [inline packages/bot/src/transformers/interaction.ts:281]:
```suggestion
```
Why is this commented?

Comment (Fleny113) [reply packages/bot/src/transformers/interaction.ts:281]: idk, it was there, it left it alone, the change is bc while doing CTRL + D in vsc it selected that too

File: packages/bot/src/transformers/interaction.ts (hunk 1 above)
Comment (Fleny113) [reply packages/bot/src/transformers/interaction.ts:244]: The guild transformers has payload checks in the file, so there shouldn't be an issue with this

File: packages/bot/src/transformers/interaction.ts (hunk 2 above)
Comment (lts20050703) [reply packages/bot/src/transformers/interaction.ts:281]: Unrelated: What's Ctrl+D?

**Review by lts20050703 — APPROVED** [review]
LGTM!

File: packages/bot/src/transformers/interaction.ts (hunk 2 above)
Comment (Fleny113) [reply packages/bot/src/transformers/interaction.ts:281]: Shortcut for Find Match in vscode

If you select a word (like "payload.") and press CTRL + D will find the next "payload." and add a cursor on that, there is also a shortcut for all matches

**Review by lts20050703 — APPROVED** [review]

**Pull request merged**